llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -44,8 +45,9 @@ Type *VPTypeAnalysis::inferScalarTypeForRecipe(const VPInstruction *R) {
     CachedTypes[OtherV] = ResTy;
     return ResTy;
   }
+  case VPInstruction::Not:
```

Comment:
`Not` should only be used to negate booleans (or vectors of bools); Would probably be better to return `IntegerType::get(Ctx, 1);` like for `FCmp/ICmp` and assert that the inferred type of the operand is also `i1`.

(The reason why `PtrAdd` doesn't return `PointerType::get()` but infers the type of the pointer arg is due to address spaces)